test/src/com/connectsdk/service/RokuServiceTest.java

```diff
@@ -0,0 +1,116 @@
+            this.response = response;
+        }
+
+        @Override
+        public void sendCommand(ServiceCommand<?> serviceCommand) {
+            mLatestCommand = serviceCommand;
+            serviceCommand.getResponseListener().onSuccess(response);
+        }
+
+    }
+
+    @Before
+    public void setUp() {
+        serviceDescription = Mockito.mock(ServiceDescription.class);
+        ServiceConfig serviceConfig = Mockito.mock(ServiceConfig.class);
+        service = new StubRokuService(serviceDescription, serviceConfig);
+    }
+
+    @Test
+    public void testServiceNotNull() {
+        Assert.assertNotNull(service);
+    }
+
+    @Test
+    public void testPlayVideo() {
+        Mockito.when(serviceDescription.getIpAddress()).thenReturn("127.0.0.1");
+        Mockito.when(serviceDescription.getPort()).thenReturn(13);
+        MediaInfo mediaInfo = new MediaInfo("url", "video/mp4", "title", "description");
+        MediaPlayer.LaunchListener listener = Mockito.mock(MediaPlayer.LaunchListener.class);
+        MediaPlayer.MediaLaunchObject response = Mockito.mock(MediaPlayer.MediaLaunchObject.class);
+        service.setResponse(response);
+        service.playMedia(mediaInfo, false, listener);
+        Assert.assertEquals("http://127.0.0.1:13/input/15985?t=v&u=url&k=(null)" +
+                "&videoName=title&videoFormat=mp4", service.mLatestCommand.getTarget());
+    }
+
+    @Test
+    public void testPlayAudio() {
+        Mockito.when(serviceDescription.getIpAddress()).thenReturn("127.0.0.1");
+        Mockito.when(serviceDescription.getPort()).thenReturn(13);
+        MediaInfo mediaInfo = new MediaInfo("url", "audio/mp3", "title", "description");
+        MediaPlayer.LaunchListener listener = Mockito.mock(MediaPlayer.LaunchListener.class);
+        MediaPlayer.MediaLaunchObject response = Mockito.mock(MediaPlayer.MediaLaunchObject.class);
+        service.setResponse(response);
+        service.playMedia(mediaInfo, false, listener);
+        Assert.assertEquals("http://127.0.0.1:13/input/15985?t=a&u=url&k=(null)" +
+                "&songname=title&artistname=description&songformat=mp3&albumarturl=(null)",
+                service.mLatestCommand.getTarget());
+    }
+
+    @Test
+    public void testDisplayImage() {
+        Mockito.when(serviceDescription.getIpAddress()).thenReturn("127.0.0.1");
+        Mockito.when(serviceDescription.getPort()).thenReturn(13);
+        MediaInfo mediaInfo = new MediaInfo("url", "image/jpeg", "title", "description");
+        MediaPlayer.LaunchListener listener = Mockito.mock(MediaPlayer.LaunchListener.class);
+        MediaPlayer.MediaLaunchObject response = Mockito.mock(MediaPlayer.MediaLaunchObject.class);
+        service.setResponse(response);
+        service.displayImage(mediaInfo, listener);
+        Assert.assertEquals("http://127.0.0.1:13/input/15985?t=p&u=url&tr=crossfade",
+                service.mLatestCommand.getTarget());
+    }
```

Comment:
The three methods are almost the same. Makes sense to extract the common stuff into a new method.